File: src/main/java/test/shihtim/HelloWorld.java
package test.shihtim;

public class HelloWorld {
    public static void main(String[] args){
        String s="abcde";
        System.out.println(s.charAt(3));
        System.out.println(s.substring(2,4));
//        System.out.println("Hello World");
        Person person=new Person("DK",60,1.7f);
//        person.name="DK";
        person.age=23;
//        person.height=1.7f;
//        person.weight=60;
        person.hello();

        System.out.println(person.bmi());
//        int age=23;
//        Integer integer=123;
//        short b=10;
//        Short ss;
//        long bc=1;
//        char cha='G';
//
//        byte ba=8;
//        boolean tr=true;
//        boolean fal=false;
//        float fol=50.5f;
//        double dob=60.43;
//        String jj="你好";
    }
}

File: src/main/kotlin/test/shihtim/HelloWorld.kt
package test.shihtim

fun main() {
//    println("Hello Kotlin");
//    Human().hello();
    var s:String?="abcde"
    s=null
    println(s?.length)
    val human=Human("DK",66.5f,1.7f);
    human.hello();
    println(human.bmi())
    var a:Int=123;
    var name:String="我不好"
}

class Human( var name:String, var weight:Float,var height:Float){
    init {
        println("test $weight ")
    }
//    constructor(name:String,weight: Float,height: Float) : this(weight, height){
//        //println(name)
//    }

    fun bmi():Float{
        val bmi=weight/(height*height)
        return bmi
    }
    fun hello(){
        println("Hello Kotlin")
    }
}